test/reduce.jl

```diff
@@ -1,305 +1,416 @@
+
+        @test maximum(abs2, 3:7) == 49
+
+        @test maximum(Int16[1]) === Int16(1)
+        @test maximum(collect(Int16(1):Int16(100))) === Int16(100)
+        @test maximum(Int32[1,2]) === Int32(2)
+    end
+
+    @testset "maxabs" begin
+        @test maxabs(Int[]) == 0
+        @test maxabs(-2) == 2
+        @test maxabs([1, -2, 3, -4]) == 4
+    end
+
+
+    @testset "minimum" begin
+        @test_throws ArgumentError minimum(Int[])
+        @test minimum(5) == 5
+        @test minimum([4, 3, 5, 2]) == 2
+
+        @test isnan(minimum([NaN]))
+        @test isnan(minimum([NaN, 2., 3.]))
+        @test isnan(minimum([4., 3., NaN, 5., 2.]))
+
+        @test minimum(abs2, 3:7) == 9
+    end
+
+    @testset "minabs" begin
+        @test_throws ArgumentError minabs(Int[])
+        @test minabs(-2) == 2
+        @test minabs([-1, 2, -3, 4]) == 1
+    end
+
+    @testset "extrema" begin
+        @test extrema(5) == (5, 5)
+        @test extrema([4, 3, 5, 2]) == (2, 5)
+        @test isequal(extrema([NaN]), (NaN, NaN))
+        @test isequal(extrema([NaN, 2., 3.]), (NaN,NaN))
+        @test isequal(extrema([4., 3., NaN, 5., 2.]), (NaN,NaN))
+
+        @test extrema(reshape(1:24,2,3,4),1) == reshape([(1,2),(3,4),(5,6),(7,8),(9,10),(11,12),(13,14),(15,16),(17,18),(19,20),(21,22),(23,24)],1,3,4)
+        @test extrema(reshape(1:24,2,3,4),2) == reshape([(1,5),(2,6),(7,11),(8,12),(13,17),(14,18),(19,23),(20,24)],2,1,4)
+        @test extrema(reshape(1:24,2,3,4),3) == reshape([(1,19),(2,20),(3,21),(4,22),(5,23),(6,24)],2,3,1)
+    end
 end
 
-@test isequal(cummin([1, 2, 5, -1, 3, -2]), [1, 1, 1, -1, -1, -2])
-@test isequal(cummax([1, 2, 5, -1, 3, -2]), [1, 2, 5, 5, 5, 5])
+immutable SomeFunctor end
+(::SomeFunctor)(x) = true
 
-@test isequal(cummax([1 0; 0 1], 1), [1 0; 1 1])
-@test isequal(cummax([1 0; 0 1], 2), [1 1; 0 1])
-@test isequal(cummin([1 0; 0 1], 1), [1 0; 0 0])
-@test isequal(cummin([1 0; 0 1], 2), [1 0; 0 0])
+@testset "logical" begin
+    @testset "any" begin
+        @test any([]) == false
+        @test any(Bool[]) == false
+        @test any([true]) == true
+        @test any([false, false]) == false
+        @test any([false, true]) == true
+        @test any([true, false]) == true
+        @test any([true, true]) == true
+        @test any([true, true, true]) == true
+        @test any([true, false, true]) == true
+        @test any([false, false, false]) == false
+
+        @test any(x->x>0, []) == false
+        @test any(x->x>0, Int[]) == false
+        @test any(x->x>0, [-3]) == false
+        @test any(x->x>0, [4]) == true
+        @test any(x->x>0, [-3, 4, 5]) == true
+
+        @test_throws TypeError any(x->0, [false])
+
+        @testset "short-circuiting" begin
+            let c = 0, A = 1:1000
+                any(x->(c=x; x==10), A)
+                @test c == 10
+            end
+        end
+
+        @testset "functors" begin
+            @test all(SomeFunctor(), 1:10)
```

Comment:
This is strange: here your comment is shown on line 307, but after my change, when I enter in "File changed", it is on line 339. I believe that's because I squashed the correction into the second commit of this PR. Anyway, the problem was solved.